### ISSUE TITLE
Assertion failure: HaveAffineDeltas false in ComputeSearchDirection

### ISSUE THREAD

**Issue by svigerske** [coin-or/Ipopt#114]
Issue created by migration from Trac.

Original creator: guest

Original creation time: 2009-12-14 09:38:18

Assignee: ipopt-team

Version: 3.8

CC:  dave@meantradingsystems.com

Sometimes HaveAffineDeltas is false on line 84 of IpPDSearchDirCalc.cpp, part of the Mehrotra algorithm.

**Comment by svigerske** [comment]
Comment by @andrea5w created at 2010-04-27 16:25:45

Sorry for having taking so long to respond to this ticket.  Can you provide information so that I can reproduce the problem? (e.g., an AMPL example?)

**Comment by svigerske** [comment]
Comment by @andrea5w created at 2010-04-27 16:25:45

Changing assignee from ipopt-team to @andrea5w.

**Comment by svigerske** [comment]
Comment by @andrea5w created at 2010-04-28 15:51:13

(this was sent to Ipopt-tickets`@`list.coin-or.org)

```
Hi,No problem about taking a while to reply - I guess you have many other things to do.

I'm afraid I'm restricted in what I can say due to non-disclosure restrictions, but I'll tell
you what I can, and what I can recall - we've been using a work-around for some time, changing
line 82 of IpPDSearchDirCalc.cpp to check �HaveAffineDeltas(); my understanding of what was in
that branch of the if statement was that it was an optimisation, so could be excluded.

We have written a program which uses Ipopt from C++, so I can't give you an AMPL example. The
problem in question is a quadratic problem with linear constraints, and we use the mumps
solver. I believe it occurred when the problem happened to only have a single feasible solution
- all the variables are constrained to zero.

I've set the following options:
jac_c_constant�yes
jac_d_constant�yes
hessian_constant�yes
mehrotra_algorithm�yes
and the following ones which I assume will make little difference:
print_level to 12
check_derivatives_for_naninf
derivative_test to second_order
print_timing_statistics yes

I hope this helps, if I can do anything else just drop me a mail - I can reproduce it here,
Dave.

PS Are you aware that the "function evaluation" timing statistics appear not to reset between
runs, whereas the others do, leading to some slightly confusing output.
```

**Comment by svigerske** [comment]
Comment by @svigerske created at 2012-11-19 16:32:13

It seems safe to me to just check for `HaveAffineDeltas()`.
This is b045127108, will be merged into stable/3.10, and will be in the next release.

Remaining issue: Are you aware that the "function evaluation" timing statistics appear not to reset between runs, whereas the others do, leading to some slightly confusing output.

**Comment by svigerske** [comment]
Comment by @svigerske created at 2012-11-19 17:05:10

Issue with resetting timing statistics was fixed with 567f9b9bbb.

**Comment by svigerske** [comment]
Comment by @svigerske created at 2012-11-19 17:05:10

Resolution: fixed

**Comment by svigerske** [comment]
Comment by @svigerske created at 2012-11-19 17:52:33

I reverted b045127108, because this issue sounds very similar to the one that was fixed with 3ac1c43943 recently.
The same assert came up there, and Tony proposed the same fix, but 3ac1c43943 did a better one.
